### PR TITLE
add IsRelational extension for DatabaseFacade

### LINKED ISSUE #6983: Add IsRelational(), IsSqlServer(), IsInMemory(), etc. (or GetProviderName()) on DatabaseFacade
### Steps to reproduce

Given the following class:

```csharp
class DummyContext : DbContext
{
     protected override void OnConfiguring(DbContextOptionsBuilder builder)
     {
            builder.UseInMemoryDatabase();
     }
}
```

When used in the following manner:

````csharp
public class WittyClassName
{
    DbContext ctx;

    public WittyClassName(DbContext ctx)
    {
        this.ctx = ctx;
    }

    public async Task DoWorkAsync()
    {
        await ctx.Database.MigrateAsync();
    }
}
````

### The issue
The above code will grant the following error on the line of `MigrateAsync` when executing `DoWorkAsync`:

````
System.InvalidOperationException : Unable to resolve service for type 'Microsoft.EntityFrameworkCore.Migrations.IMigrator'. This is often because no database provider has been configured for this DbContext. A provider can be configured by overriding the DbContext.OnConfiguring method or by using AddDbContext on the application service provider. If AddDbContext is used, then also ensure that your DbContext type accepts a DbContextOptions<TContext> object in its constructor and passes it to the base constructor for DbContext.
  Stack Trace:
       at Microsoft.EntityFrameworkCore.Infrastructure.AccessorExtensions.GetService[TService](IInfrastructure`1 accessor)
       at Microsoft.EntityFrameworkCore.RelationalDatabaseFacadeExtensions.MigrateAsync(DatabaseFacade databaseFacade, CancellationToken cancellationToken)
````

This only occurs with the in memory database from my experience and works just fine with the others I've used (which is just sqlite). I would have expected this to have just "worked"; if an in memory schema cannot be created, then make the migration be a no-op.

I can resolve this by providing my own service provider but this involves having to mock not only IMigrator but any other service that the database uses, like `IServiceScopeFactory`. This is a lot of work to have an integration test in an in memory database that utilises migrations. The other option would be to leave this section of the code untested.

### Further technical details

EF Core version: `1.0.1`, `Microsoft.EntityFrameworkCore.InMemory` is also `1.0.1`.
Operating system: OSX Sierra 10.12.1
Visual Studio version: using VS Code 1.5.3

### PR DESCRIPTION
**Pull request by WeihanLi**
add `IsRelational` extension for `DatabaseFacade`

Fixes #6983 

Tests for the changes have been added (for features)

### CODE REVIEW

File: src/EFCore.Relational/Extensions/RelationalDatabaseFacadeExtensions.cs [hunk 1]
```diff
@@ -604,6 +604,19 @@ public static void SetCommandTimeout([NotNull] this DatabaseFacade databaseFacad
...
+        /// <summary>
+        ///     <para>
+        ///         Returns true if the database provider currently in use is a relational database.
+        ///     </para>
+        /// </summary>
+        /// <param name="databaseFacade"> The facade from <see cref="DbContext.Database" />. </param>
+        /// <returns> True if a relational database provider is being used; false otherwise. </returns>
+        public static bool IsRelational([NotNull] this DatabaseFacade databaseFacade)
+        {
+            var dependencies = ((IDatabaseFacadeDependenciesAccessor)databaseFacade).Dependencies;
```
Comment (ajcvickers) [inline src/EFCore.Relational/Extensions/RelationalDatabaseFacadeExtensions.cs]: Use Check.NotNull

File: src/EFCore.Relational/Extensions/RelationalDatabaseFacadeExtensions.cs [hunk 2]
```diff
@@ -604,6 +604,19 @@ public static void SetCommandTimeout([NotNull] this DatabaseFacade databaseFacad
...
+            return dependencies is IRelationalDatabaseFacadeDependencies;
```
Comment (ajcvickers) [inline src/EFCore.Relational/Extensions/RelationalDatabaseFacadeExtensions.cs]: Consider collapsing into a single line.

File: test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs [hunk 3]
```diff
@@ -23,6 +23,15 @@ namespace Microsoft.EntityFrameworkCore
 {
     public class RelationalDatabaseFacadeExtensionsTest
     {
+        [ConditionalFact]
+        public void IsRelational_returns_if_relational()
```
Comment (ajcvickers) [inline test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs]: Add a test that this returns false when using the in-memory database.

Comment (WeihanLi) [reply test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs]: As the `IsRelational` is an extension method in `EfCore.Relational`，I'm not sure where should I add the test, any suggestion?

Comment (ajcvickers) [reply test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs]: Should be able to add it here. I believe the unit tests have access to the in-memory database as well.

Comment (WeihanLi) [reply test/EFCore.Relational.Tests/RelationalDatabaseFacadeExtensionsTest.cs]: yes, I'd add the in-memory test

**Comment by ajcvickers** [comment]
@WeihanLi Thanks!

**Pull request merged**